src/options.js

```diff
@@ -9,8 +21,105 @@ import { _catchError } from './diff/catch-error';
  * ctrl+click or cmd+click on mac the type definition below).
  * @type {import('./internal').Options}
  */
-const options = {
-	_catchError
-};
+const options = {};
+
+Object.defineProperty(options, 'vnode', {
```

Comment:
I wonder if we could use a helper to define the value field and the options hook property at the same time?

```js
export let vnode = optionsHook('vnode');

/**
 * @template {keyof import('./internal').Options} T
 * @param {T} name
 * @param {any} [value] - initial value for the hook
 * @returns {import('./internal').Options[T]}
 */
function optionsHook(name, value) {
	Object.defineProperty(options, 'unmount', {
		get() {
			return value;
		},
		set(newValue) {
			value = newValue;
		}
	});
	return value;
}
```

Reply:
@developit the byte-size change is actually very minimal